### PR TITLE
994_Rotting_Oranges

### LINKED ISSUE #62: 994_Rotting_Oranges
## Summary
994_Rotting_Oranges
## Goal
994_Rotting_Oranges
## Todo

- [ ] add 994_Rotting_Oranges
- [ ] write time and space complexity

## Deadline
2/1
## Parent issue
https://github.com/okuda-seminar/Engineering/issues/599
## References
None

## Notes
None

### PR DESCRIPTION
**Pull request by Rwatana**
Please check my code.

### CODE REVIEW

File: 994_Rotting_Oranges.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                    rotted_group.append((row_num + 1, col_num))
+                if row_num - 1 >= 0 and grid[row_num - 1][col_num] == 1:
+                    grid[row_num-1][col_num] = 2
+                    rotted_group.append((row_num-1, col_num))
+                if col_num + 1 < col and grid[row_num][col_num + 1] == 1:
+                    grid[row_num][col_num+1] = 2
+                    rotted_group.append((row_num, col_num + 1))
+                if col_num - 1 >= 0 and grid[row_num][col_num - 1] == 1:
+                    grid[row_num][col_num - 1] = 2
+                    rotted_group.append((row_num, col_num - 1))
```
Comment (ryujumoto) [inline 994_Rotting_Oranges.py]: You can compress these `if` blocks into one `for` loop. Refer to [this code](https://github.com/okuda-seminar/leetcode_answers/blob/main/994_rotting_oranges.py).

File: 994_Rotting_Oranges.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Time: O(n**2)
+# Space: O(1)
+
+
+class Solution(object):
+    def orangesRotting(self, grid):
```
Comment (ryujumoto) [inline 994_Rotting_Oranges.py]: Add type annotation.

File: 994_Rotting_Oranges.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Time: O(n**2)
+# Space: O(1)
```
Comment (ryujumoto) [inline 994_Rotting_Oranges.py]: Time complexity and space one are wrong.

**Review by ryujumoto — COMMENTED** [review]
Done.

**Comment by Rwatana** [comment]
I understood how to make the code easier. Please check.

**Pull request closed without merging**